presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -122,7 +122,9 @@ private Type instantiateParametricType(TypeSignature signature)
     {
         ImmutableList.Builder<Type> parameterTypes = ImmutableList.builder();
         for (TypeSignature parameter : signature.getParameters()) {
-            parameterTypes.add(getType(parameter));
+            Type parameterType = getType(parameter);
+            checkNotNull(parameterType, "Invalid type: %s", parameter.toString());
```

Comment:
Should this return `null` here so that the caller can provide the appropriate error?